shim/Cargo.toml

```diff
@@ -20,7 +20,12 @@ anyhow = { version = "1.0.65", features = ["backtrace"] }
 clap = { version = "3.2.23", features = ["derive"] }
 globset = "0.4.9"
 predicates = "2.1.1"
+# semver does note that it is designed to parse Cargo's interpretation of semver.
+# Therefore we cannot use it to parse semver strings in general. However, it is
+# good enough to parse semver strings that we control, i.e. turbo's own version
+semver = "1.0.14"
 serde = { version = "1.0.145", features = ["derive"] }
 serde_json = "1.0.86"
 serde_yaml = "0.8.26"
 walkdir = "2"
+yarn-lock-parser = "0.4.0"
```

Comment:
Took a quick look at this parser and I don't love it. It uses the same parser for Yarn 1 and 2+ when they are different formats and even Yarn has different parsers for each format. It also gets package name by looking at the first chunk of whatever is the first descriptor for an entry and not the locator.

I think it should be fine for the this use case, but if we need any more info from the yarn lockfile we should switch it up.